### DISCUSSION TITLE
How to map key from current folder

### DISCUSSION THREAD

**Discussion by Neowuf** [caronchen/obsidian-chartsview-plugin#64] [Q&A] [answered]
Visually, this is what I want to achieve.
I can do that using charts view template, but I want to use dataviewJS.
![image](https://github.com/user-attachments/assets/39709067-30a0-4159-baa6-339e65927680)

_ _ _
The key can be static. But I want the value to update when I change the number in the property.
I try. my logic dont work. Please help.
![image](https://github.com/user-attachments/assets/7d01bc1c-1446-4128-82a4-570254d78c92)

**Comment by nelsonbianchini** [comment]
Did you find any way to do that?

**Reply by caronchen** [reply, marked as answer]
@nelsonbianchini @Neowuf 

<img width="753" height="408" alt="image" src="https://github.com/user-attachments/assets/c209c816-05cb-4b18-84a1-fecac2504392" />


```chartsview
#-----------------#
#- chart type    -#
#-----------------#
type: Bar

#-----------------#
#- chart data    -#
#-----------------#
data: |
  dataviewjs:
  const loadData = async () => {
      return dv.pages()
               .groupBy(p => p.file.folder.split('/')[0])
               .map(p => ({directory: p.key === '' ? '/':p.key, count: p.rows.length}))
               .array();
  };
  return await loadData();

#-----------------#
#- chart options -#
#-----------------#
options:
  xField: "count"
  yField: "directory"
  label:
    position: "middle"
```
